replicatedhq/ship
# move to versioned state

What I Did
------------

Prepare to expand the things we store in state by adding some more keys to state.json instead of just raw customer config

How I Did it
------------

- when reading state.json, we check if it has a `v1.config`, and if so, we assume its a `VersionedState`
- otherwise we assume its a raw `map[string]interface{}` containg plain `ConfigOption` values

How to verify it
------------

Should be no visible changes, can check we don't break anything by ensuring you can read in a v0/raw state.json and still get
your previous values.

Description for the Changelog
------------

Files stored at `.ship/state.json` will now be written in a new format. State files written by earlier versions of ship will be transparently translated to the new format the next time Ship is run.

![](https://thumbs.dreamstime.com/t/two-boats-lake-birch-hdr-48738892.jpg)
